### DISCUSSION TITLE
Cannot make a moosebased App on a linux cluster

### DISCUSSION THREAD

**Discussion by morsi95** [idaholab/moose#23825] [Q&A Getting Started] [answered]
I have got the following error when I try to make a moose App on a linux cluster:
```
Linking Library /dss/dsshome1/lxc0F/ge45dal2/golem/lib/libgolem-opt.la...
Linking Executable /dss/dsshome1/lxc0F/ge45dal2/golem/golem-opt...
/usr/bin/ld: /dss/dsshome1/lxc0F/ge45dal2/golem/test/lib/libgolem_test-opt.so: undefined reference to `InputParameters validParams<GolemApp>()'
collect2: error: ld returned 1 exit status
/dss/dsshome1/lxc0F/ge45dal2/offline/moose/framework/app.mk:415: recipe for target '/dss/dsshome1/lxc0F/ge45dal2/golem/golem-opt' failed
make: *** [/dss/dsshome1/lxc0F/ge45dal2/golem/golem-opt] Error 1
```
I have created a moose profile where i can source moose each time I start a shell and I ran it before trying to make the App. This is the moose_profile file:
```
module unload devEnv/Intel/2019
module load gcc/9 mpi.intel/2019_gcc mkl/2019_gcc python/3.6_intel
export PATH=/lrz/sys/intel/impi2019u6/impi/2019.6.154/intel64/bin/:/lrz/sys/compilers/gcc/9.2/bin:$PATH
export CC=mpicc CXX=mpicxx FC=mpif90 F90=mpif90 F77=mpif77
export PETSC_DIR=$HOME/libs/petsc
export LIBMESH_DIR=$HOME/libs/libmesh
export MOOSE_DIR=$HOME/offline/moose

```

**Comment by GiudGiud** [comment, marked as answer]
Hello

Sorry we missed this we do not handle user support in Issues, we handle it in github Discussions. 

This particular error was caused by the deprecation of the old form of validParams in MOOSE about a year ago. 
Please follow the instructions here to fix your application. Every object's validParam will have to be updated
https://mooseframework.inl.gov/newsletter/2020/2020_04.html#c1ba1ef6-13a6-4210-bedb-0b691fec2ce7

Best,
Guillaume
